CIP-0072/version_1_offchain.json

```diff
@@ -11,63 +16,92 @@
-          "enum":["Games", "DeFi", "Gambling", "Exchanges", "Collectibles", "Marketplaces", "Social", "Other"]
+          "enum": ["DeFi", "Development", "Education", "Games", "Identity", "Marketplace", "NFT", "Other", "Security"]
         },
         "description": "One or more categories. Category MUST be one of the following schema definition."
       },
+      "screenshots": {
+        "type": "array",
+        "maxItems": 10,
+        "items": {
+          "type": "string",
+          "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})",
+          "maxLength": 1000,
+          "description": "Minimum resolution: 1920x1080px, supported formats: PNG/JPG/SVG, maximum file size: 1 MB"
+        },
+        "description": "URLs to screenshots of DApp actual pages. We recommend to share screenshots from the dApp usage itself."
+      },
       "social": {
         "type":"object",
         "properties": {
-          "website": {
-            "type":"string",
-            "description": "dApps website link.",
+          "companyName": {
+            "type": "string",
+            "description": "Name of the company developing/maintaining the dApp",
+            "maxLength": 50
+          },
+          "companyEmail": {
+            "type": "string",
+            "description": "Email of the company developing/maintaining the dApp",
+            "maxLength": 150,
+            "format": "email"
+          },
+          "companyWebsite": {
+            "type": "string",
+            "description": "Website of the company developing/maintaining the dApp",
+            "maxLength": 200,
             "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
           },
           "twitter": {
             "type":"string",
             "description": "An optional Twitter link.",
+            "maxLength": 200,
             "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
           },
           "github": {
             "type":"string",
             "description": "An optional Github link.",
+            "maxLength": 200,
             "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
           },
           "discord": {
             "type":"string",
             "description": "An optional Discord link.",
+            "maxLength": 200,
```

Comment:
What is the benefit by limiting this, if this is offchain?

Reply:
@marcin-mazurek I'd also like to know why this was added.  From experience I know in the vast majority of cases it's a sensible limit, and also that the hard limit will make _something_ fail at some future time.  What is the actual risk for these values to have an arbitrary & potentially huge length?